census/src/test/scala/grasshopper/census/util/Generators.scala

```diff
@@ -0,0 +1,36 @@
+package grasshopper.census.util
+
+import org.scalacheck.{ Arbitrary, Gen }
+
+trait Generators {
+
+  case class Alphanumeric(s: String) {
+    override def toString = s.toString
+  }
+
+  case class HyphenStr(s: String) {
+    override def toString = s.toString
+  }
+
+  val unicodeChar = Gen.choose(Char.MinValue, Char.MaxValue).filter(c => !Character.isDigit(c))
+
+  val alphanumericGen: Gen[Alphanumeric] = {
+    for {
+      x <- Gen.oneOf("A", "B", "C")
+      y <- Gen.choose(Int.MinValue, Int.MaxValue)
```

Comment:
Do you want to allow negative numbers?  You're going to end up with both `A1` and `A-1` type values here.